controllers/ebpf/agent_controller.go

```diff
@@ -160,6 +175,30 @@ func (c *AgentController) desired(ctx context.Context, coll *flowslatest.FlowCol
 	if err != nil {
 		return nil, err
 	}
+	volumeMounts := c.volumes.GetMounts()
+	volumes := c.volumes.GetVolumes()
+
+	if helper.IsTCPDropEnabled(&coll.Spec) || helper.IsDNSTrackingEnabled(&coll.Spec) {
+		if !coll.Spec.Agent.EBPF.Privileged {
+			return nil, fmt.Errorf("to use TCPDrop feature privileged mode need to be enabled")
```

Comment:
Seems a bit harsh to return an error: it will make reconciliation failing in loop until the user fixes it, so nothing would be usable, as the agent pods won't be deployed. It could be a strategy to force users fix the issue (they'll have to look at the operator logs to figure it out) - but still I find it harsh.

I see two alternatives:
- Just log the error instead and proceed as if the two features were disabled (ideally we should also propagate some warning into the CR status/conditions - could be a follow-up, I'm not sure if we implement that already)
- Or make these features also turn on the privileged mode, automatically. This is probably more user-friendly, but we should be crystal-clear about that in the documentation.

any thoughts?

Reply:
I originally was enabling privileged under the hood but changed that as I don't think this is good practice its better to be explicit with the configs rather that implicit and we also call that clearly in the doc if u enable those features u have to enable privileged otherwise it won't work so we can log and err/warning and proceed ?